ext/src/js/modules/main.js

```diff
@@ -16,200 +16,206 @@ import {
-     * Event handlers                   *
-     * ---------------------------------*/
+    // fill url input field
+    document.getElementById('url').value = getSiteUrl();
 
-    function backHandler() {
-        clearContentContainer();
-        updateBoundVar([{ 'onoptions': false }, { 'onafterdownload': false }, { 'onmain': true }]);
-        fillFormFields();
+    // fill title input field
+    const title = await executeScriptToPromise(() => document.title);
+    if (title && title[0]) {
+        document.getElementById('title').value = title[0];
     }
 
-    async function optionsHandler() {
-        updateBoundVar([{ 'onoptions': true }, { 'onmain': false }]);
-
-        const optionsPageText = await fetch('./options.html').then(stream => stream.text());
-        const p = new DOMParser();
-        const optionsPageElement = p.parseFromString(optionsPageText, 'text/html');
-        const template = optionsPageElement.querySelector('template')?.content.cloneNode(true);
-        const script = optionsPageElement.querySelector('script');
-        contentContainer.appendChild(template);
-        contentContainer.appendChild(copyScript(script));
+    // fill notes input field
+    const selection = await executeScriptToPromise(() => window.getSelection().toString());
+    if (selection && selection[0]) {
+        document.getElementById('notes').value = selection[0];
     }
 
-    function closeHandler() {
-        window.close();
+    //fill tags
+    tagInput.renderTags();
+
+}
+
+/* ---------------------------------*
+    * Event handlers                   *
+    * ---------------------------------*/
+
+function backHandler() {
+    clearContentContainer();
+    updateBoundVar([{ 'onoptions': false }, { 'onafterdownload': false }, { 'onmain': true }]);
+    fillFormFields();
+}
+
+async function optionsHandler() {
+    updateBoundVar([{ 'onoptions': true }, { 'onmain': false }]);
+
+    const optionsPageText = await fetch('./options.html').then(stream => stream.text());
+    const p = new DOMParser();
+    const optionsPageElement = p.parseFromString(optionsPageText, 'text/html');
+    const template = optionsPageElement.querySelector('template')?.content.cloneNode(true);
+    const script = optionsPageElement.querySelector('script');
+    contentContainer.appendChild(template);
+    contentContainer.appendChild(copyScript(script));
+}
+
+function closeHandler() {
+    window.close();
+}
+
+/* ---------------------------------*
+    * Save bookmarks                   *
+    * ---------------------------------*/
+
+async function createBookmark(e) {
+    e.preventDefault();
+    form = new FormData(document.forms['add']);
+    updateBoundVar([{ 'onafterdownload': true }, { 'onmain': false }]);
+    renderProgressBar(document.getElementById('omnom_status'));
+    createSnapshot();
+}
+async function saveBookmark(snapshotData) {
+    form.set('tags', tagInput.getTags().join(','));
+
+    //console.time('createSnapshot');
```

Comment:
It's been nice to see the snapshotting performance, but if we don't need, could you please delete the commented console calls?